File: Q30/Q30.py
from collections import Counter

n = int(input("Enter the size of the array: "))
arr = list(map(int, input("Enter the elements of the array: ").split()))

frequency = Counter(arr)


most_frequent = max(frequency, key=frequency.get)
max_count = frequency[most_frequent]

print(f"The most frequent element is {most_frequent} and it appears {max_count} times.")

File: Q30/Q30.go
package main

import (
	"fmt"
)

func mostFrequentElement(arr []int) (int, int) {
	frequency := make(map[int]int)
	maxCount := 0
	mostFrequent := arr[0]

	for _, num := range arr {
		frequency[num]++
		if frequency[num] > maxCount {
			maxCount = frequency[num]
			mostFrequent = num
		}
	}

	return mostFrequent, maxCount
}

func main() {
	arr := []int{1, 3, 2, 1, 4, 1, 3, 2, 3, 3}
	element, count := mostFrequentElement(arr)
	fmt.Printf("Most frequent element: %d (occurs %d times)\n", element, count)
}
